dashboard/test/models/contact_rollups_processed_test.rb

```diff
@@ -0,0 +1,130 @@
+require 'test_helper'
+
+class ContactRollupsProcessedTest < ActiveSupport::TestCase
+  test 'import_from_raw_table creates one row per email' do
+    clean_tables
+
+    # Create 6 records for 3 unique emails
+    create :contact_rollups_raw, email: 'email1@example.domain'
+    create_list :contact_rollups_raw, 2, email: 'email2@example.domain'
+    create_list :contact_rollups_raw, 3, email: 'email3@example.domain'
+
+    ContactRollupsProcessed.import_from_raw_table
+
+    # Note: having unique email addresses is already guaranteed by table constraint
+    assert_equal 3, ContactRollupsProcessed.count
+  end
+
+  test 'import_from_raw_table inserts records by batch' do
+    unique_email_count = 15
+    batch_sizes = [1, 5, 7, 11, 20]
```

Comment:
Why so many options? I see doing 1, something divisible by the total number of rows (5), something not divisible by the total (7), and something bigger than the total (20), so I guess that you only have one more than that.

Reply:
My thought was
5 = does many imports inside the loop but none outside. (Batch size of 1 is the extreme case of this kind.)
7 = does more than 1 imports inside the loop, 1 outside.
11 = does just 1 import inside the loop, 1 outside.
20 = no import in the loop, 1 outside.